src/objects/job.py

```diff
@@ -472,3 +479,99 @@ def _find_test_failures(self, junit_dir: str) -> list[Failure]:
             )
 
         return failures
+
+    def _get_timestamp(
+        self,
+        job_name: Optional[str],
+        build_id: Optional[str],
+    ) -> Optional[int]:
+        """
+        Fetches the UNIX timestamp of job
+        
+        Args:
+            job_name (Optional[str]): The name of the job to get timestamp for.
+            build_id (Optional[str]): The build ID of the job to get timestamp for.
+
+        Returns:
+            Optional[int]: The timestamp if found, otherwise None.
+        """
+
+        client = storage.Client()
+        bucket = client.bucket(self.gcs_bucket)
+        blob = bucket.blob(f"{job_name}/{build_id}/started.json")
+
+        try:
+            started_json = json.loads(blob.download_as_text())
+            return started_json.get("timestamp")
+        except Exception as e:
+            self.logger.error(f"Failed to fetch timestamp from started.json: {e}")
+            return None
+
+    def _get_all_build_ids(
+        self,
+        job_name: Optional[str],
+    ) -> list[str]:
+        """
+        Get the list of all build IDs from the job directory.
+        
+        Args:
+            job_name (Optional[str]): The name of the job to get timestamp for.
+    
+        Returns:
+            List[str]: A list of build IDs.
+        """
+        client = storage.Client()
+        bucket = client.bucket(self.gcs_bucket)
+        prefix = f"logs/{job_name}/"
+        blobs = bucket.list_blobs(prefix=prefix, delimiter='/')
+
+        build_ids = []
+        for page in blobs.pages:
+            build_ids.extend(page.prefixes)
+
+        # Get the build IDs from the prefixes
+        build_ids = [prefix.split('/')[-2] for prefix in build_ids if prefix.endswith('/')]
+        self.logger.info(f"Build IDs retrieved for {job_name}: {build_ids}")
+        return build_ids
+
+    def _check_is_retriggered(
+        self,
+        job_name: Optional[str],
+        build_id: Optional[str],
+    ) -> bool:
+        """Check if the current job build is retriggered within the same week as a previous build.
+
+        Args:
+            job_name (Optional[str]): The name of the job to get timestamp for.
+            build_id (Optional[str]): The build ID of the job to get timestamp for.
+
+        Returns:
+            bool: True if retriggered within the same week, False otherwise.
+        """
+        current_timestamp = self._get_timestamp(job_name, build_id)
+        if current_timestamp is None:
+            self.logger.error(f"Failed to get timestamp for job {job_name} with build ID {build_id}")
```

Comment:
I think you repeat a similar log message in the `_get_timestamp` function already:

```
self.logger.error(f"Failed to fetch timestamp from started.json: {e}")
```

Could just leave the log message in `_get_timestamp` and remove this one.

Reply:
Thanks I have remote duplicates and updated the log message to include job details in `_get_timestamp` one